### ISSUE TITLE
return value of `save_result`

### ISSUE THREAD

**Issue by soxofaan** [Open-EO/openeo-processes#279]
https://github.com/Open-EO/openeo-processes/blob/60fb5e22579f34010de42af71c3d2d70f2dea906/save_result.json#L43-L48

return value of `save_result` is true or false, but if save_result fails, a backend will probably throw a descriptive error before returning false, so nobody is going to check the actual return value.

Wouldn't it be more useful to just return (pass through) the input data so that it can be re-used for further processing?
This would allow for example:
- store  intermediate results in a batch job
- store result in multiple formats without having to re-execute same job

**Comment by soxofaan** [comment]
> This is totally valid, but you are saying it is not necessary?

that's a valid process graph, but is a back-end forced to "execute" nodes that do not contribute to the result?

**Comment by m-mohr** [comment]
Yes, I think so. The API says "multiple end nodes are possible".

**Comment by soxofaan** [comment]
(sorry for delay)

> Yes, I think so. The API says "multiple end nodes are possible". 

It currently says indeed
> One of the nodes  .. (the final one) MUST have the result flag set to true, ... Having such a node is important as multiple end nodes are possible, 

There s is some room for interpretation here.  It doesn't explicitly say that a backend must fully evaluate non-result end nodes. Analogy: a C or Java program can have multiple functions, but only `main` is triggered when the program is executed and there is no guarantee that all code will be visited.

The obligation to execute all end nodes instead of just the `"result": True` node has quite a big impact:
- the process graph parsing and (spark based) execution in VITO driver heavily depends  on  the assumption that there is only one final node to evaluate
- The data cube building approach in the python client using chained methods on a "DataCube" object also depends on that: you call `.execute()` (or its batch variant) on a "DataCube", which is by design a reference to a single node (the result node). The current machinery in the Python client does not support building or executing a process graph with multiple result nodes.

I think it should be made more explicit in the API description how to handle a process graph with multiple end nodes. 
From the above it should be clear that I favor the "only execute the result node" model. I think it's a simpler model, more straightforward to implement in both clients and back-ends and allows cleaner client API's for the end user.

Note that it still possible to "emulate" multiple result nodes in this model: you just collect all your result nodes in a final `array_create` process that acts as single result node. If necessary it could be handy to define a dedicated process for this (e.g. `collect`) at the API level  or just provide a helper at the client level.

**Comment by m-mohr** [comment]
> (sorry for delay)

No worries. I migrated over to https://github.com/Open-EO/openeo-api/issues/427
